lucille-core/src/test/java/com/kmwllc/lucille/core/WorkerTest.java

```diff
@@ -0,0 +1,75 @@
+package com.kmwllc.lucille.core;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.mockito.MockedConstruction;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import com.kmwllc.lucille.message.WorkerMessenger;
+import com.kmwllc.lucille.message.WorkerMessengerFactory;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+public class WorkerTest {
+
+  @Test
+  public void testMain() {
+    Map<WorkerPool, List<Object>> args = new HashMap<>();
+    Config config = ConfigFactory.load("WorkerTest/config.conf");
+    WorkerMessengerFactory mockFactory = new WorkerMessengerFactory() {
+      @Override
+      public WorkerMessenger create() {
+        throw new UnsupportedOperationException("Unimplemented method 'create'");
+      }
+    };
+    WorkerMessengerFactory mockFactory2 = new WorkerMessengerFactory() {
+      @Override
+      public WorkerMessenger create() {
+        throw new UnsupportedOperationException("Unimplemented method 'create'");
+      }
+    };
+    try (MockedStatic<ConfigFactory> configFactory = Mockito.mockStatic(ConfigFactory.class);
+        MockedConstruction<WorkerPool> workerPool = Mockito.mockConstruction(WorkerPool.class, (mock, context) -> {
+          args.put(mock, new ArrayList<>(context.arguments()));
+        });
+        MockedStatic<WorkerMessengerFactory> factory = Mockito.mockStatic(WorkerMessengerFactory.class)) {
+
+      configFactory.when(ConfigFactory::load).thenReturn(config);
+
+      factory.when(() -> {
+        WorkerMessengerFactory.getKafkaFactory(config, "foo");
+      }).thenReturn(mockFactory);
+
+      factory.when(() -> {
+        WorkerMessengerFactory.getKafkaFactory(config, "foo2");
+      }).thenReturn(mockFactory2);
+
+      try {
+        Worker.main(new String[0]);
+        Worker.main(new String[] {"foo"});
+      } catch (Exception e) {
+        // we do not want an exception to be thrown
+        assertTrue(false);
```

Comment:
the test will automatically fail if an exception is thrown, so we don't really need the try catch
if there's a reason why you want to be explicit, could use fail() instead of assertTrue(false)
could also use assertDoesNotThrow (need to check if it's available in the junit version we're on)
suggest the implicit approach of just letting the exception bubble up, but could add a comment with any info about what exception might possibly be thrown that we don't want

Reply:
Looks like this has been addressed in KafkaMessengerTest -- thx -- but not here?

Reply:
done